Client/Frontend/Fakespot/Client/FakeSpotClient.swift

```diff
@@ -0,0 +1,77 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+
+protocol FakeSpotClientType {
+    func fetchProductAnalysisData(productId: String, website: String) async throws -> ProductAnalysisData
+    func fetchProductData(productId: String, website: String) async throws -> [ProductAdsData]
+}
+
+struct MockFakeSpotClient: FakeSpotClientType {
+    func fetchProductAnalysisData(productId: String, website: String) async throws -> ProductAnalysisData {
+        try load(ProductAnalysisData.self, filename: "productanalysis-response")
+    }
+
+    func fetchProductData(productId: String, website: String) async throws -> [ProductAdsData] {
+        try load([ProductAdsData].self, filename: "productadsdata-response")
+    }
+
+    private func load<T: Decodable>(_ type: T.Type, filename: String) throws -> T {
+        let path = Bundle.main.url(forResource: filename, withExtension: "json")!
+        let data = try Data(contentsOf: path)
+        return try JSONDecoder().decode(type, from: data)
+    }
+}
+
+struct StagingFakeSpotClient: FakeSpotClientType {
+    func fetchProductAnalysisData(productId: String, website: String) async throws -> ProductAnalysisData {
+        // Define the API endpoint URL
+        let endpointURL = URL(string: "https://staging-trustwerty.fakespot.io/api/v1/fx/analysis")!
```

Comment:
This should probably fail more gracefully in case the URL cannot be created.

Reply:
My opinion is it shouldn't fail since it's hardcoded by the developer. It should crash immediately in debug, unless it's coming externally from nimbus.

Reply:
@adudenamedruby is working on a Swiftlint rule to get rid of force unwraps. Let's not add more work on him and just get rid of it now.

Reply:
Guard seems like a good option to solve this. But there might be other options depending on the specific code. We might disable a rule temporarily for a specific code path if there is no other way around. Here however, using guard seems straight-forward.

Reply:
Okay, I will add it